text_editor.py

```diff
@@ -1,35 +1,77 @@
 """Text Editor"""
 
-# import keyboard
+# import curses
 
 class Model:
     """
     The model for the text editor.
     API supports up/down/left/right movement of cursor and insert/delete.
     TODO: Add highlighting.
+    TODO: Read/Write files.
     """
-    def __init__(self, cursor_position=0, prev_id=None,
-                 next_id=None, curr_id=0, iterator=1) -> None:
+    def __init__(self, cursor_position=0, prev_id=None, next_id=None,
+                 curr_id=0, iterator=1, window=(10,16)) -> None:
         self.text = []
+        self.cache = {}
```

Comment:
Also, what does the cache cache?

Reply:
The cache stores the text, the previous line id, and the next line id. I'll write it in a comment